src/helpers.ts

```diff
@@ -539,88 +538,86 @@ export function updateWithdrawalAmount(
   tokenId: string,
   amount: BigInt,
 ): void {
-  let tokenBalanceId = createJointId([accountId, tokenId])
+  // Create TokenWithdrawal if account is a user
+  if (getUser(accountId)) {
+    let tokenBalanceId = createJointId([accountId, tokenId])
 
-  let tokenWithdrawalId = createJointId([
-    TOKEN_WITHDRAWAL_USER_PREFIX,
-    tokenBalanceId,
-  ])
-  let tokenWithdrawal = TokenWithdrawal.load(tokenWithdrawalId)
-  if (!tokenWithdrawal) {
-    tokenWithdrawal = new TokenWithdrawal(tokenWithdrawalId)
-    tokenWithdrawal.account = accountId
-    tokenWithdrawal.token = tokenId
-    tokenWithdrawal.amount = ZERO
+    let tokenWithdrawalId = createJointId([
+      TOKEN_WITHDRAWAL_USER_PREFIX,
+      tokenBalanceId,
+    ])
+    let tokenWithdrawal = TokenWithdrawal.load(tokenWithdrawalId)
+    if (!tokenWithdrawal) {
+      tokenWithdrawal = new TokenWithdrawal(tokenWithdrawalId)
+      tokenWithdrawal.account = accountId
+      tokenWithdrawal.token = tokenId
+      tokenWithdrawal.amount = ZERO
+    }
+    tokenWithdrawal.amount += amount
+    tokenWithdrawal.save()
   }
-  tokenWithdrawal.amount += amount
-  tokenWithdrawal.save()
-
-  // If contractId == accountId this is a split distribution, can ignore. It's a weird edge case,
-  // really shouldn't be captured in withdrawn amount but need to handle for legacy.
-  if (contractId != accountId) {
-    // Ideally can set contract earnings for any account type eventually. For now only setting them on users
-    let account = getUser(accountId)
-    if (!account) return
-
-    if (contractId) {
-      // Funds were pushed directly to the recipient, did not go through split main
-      let contractEarningsId = saveContractEarnings(contractId, accountId)
-      let contractEarningsWithdrawalId = createJointId([
-        CONTRACT_EARINGS_WITHDRAWAL_PREFIX,
-        contractEarningsId,
+
+  // Modify ContractEarnings
+  if (contractId && contractId != accountId) {
+    // Funds were pushed directly to the recipient, did not go through split main
+    let contractEarningsId = saveContractEarnings(contractId, accountId).id
+    let contractEarningsWithdrawalId = createJointId([
+      CONTRACT_EARINGS_WITHDRAWAL_PREFIX,
+      contractEarningsId,
+      tokenId,
+    ])
+    let contractEarningsWithdrawal = ContractEarningsWithdrawal.load(
+      contractEarningsWithdrawalId,
+    )
+    if (!contractEarningsWithdrawal) {
+      contractEarningsWithdrawal = new ContractEarningsWithdrawal(
+        contractEarningsWithdrawalId,
+      )
+      contractEarningsWithdrawal.contractEarnings = contractEarningsId
+      contractEarningsWithdrawal.token = tokenId
+      contractEarningsWithdrawal.amount = ZERO
+    }
+    contractEarningsWithdrawal.amount += amount
+    contractEarningsWithdrawal.save()
+  } else if (contractId == accountId || !contractId) {
+    // Funds were pushed throughs split main
+    // contractId == accountId => split distribution, !contractId => split main withdrawal
```

Comment:
I think if `contractId == accountId`, we want to skip this part. That means it's a split distribution, right? So we're not actually performing a withdrawal.

Really the split distribution shouldn't be calling this code. It's a weird leftover thing we never got rid of for legacy reasons. Was a bad design choice in the beginning.

Reply:
It logically seems a bit strange, but I think we want this?

This is part of the new changes

Split contracts will now have their own contractEarnings (which tracks balances a split earns from other splits/contracts) and distributing a split should move the balance from `internalBalance` => `withdrawals`

Reply:
Yes, got it now. ok looks good.